pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -94,17 +95,51 @@ func (pr *PipelineRun) IsGracefullyStopped() bool {
 
 // GetTimeout returns the the applicable timeout for the PipelineRun
 func (pr *PipelineRun) GetTimeout(ctx context.Context) time.Duration {
-	// Use the platform default if no timeout is set
-	if pr.Spec.Timeout == nil && pr.Spec.Timeouts == nil {
-		defaultTimeout := time.Duration(config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes)
-		return defaultTimeout * time.Minute
-	}
-
 	if pr.Spec.Timeout != nil {
 		return pr.Spec.Timeout.Duration
 	}
+	if pr.Spec.Timeouts != nil {
+		if pr.Spec.Timeouts.Pipeline != nil {
```

Comment:
nit: Could be `if pr.Spec.timeouts != nil && if pr.Spec.Timeouts.Pipeline != nil` to remove one if 😛